hipcheck/src/main.rs

```diff
@@ -503,12 +506,42 @@ fn cmd_plugin(args: PluginArgs) {
 		working_dir: working_dir.clone(),
 		entrypoint: entrypoint2.display().to_string(),
 	};
+	let exec_config = if let Some(p) = config.exec() {
+		ExecConfig::from_file(p)
+			.context("Failed to load the exec config. Please make sure the exec config file is in the provided location and is formatted correctly.")
+	} else {
+		let exec_file = "Exec.kdl";
+		let mut curr_dir= env::current_dir().unwrap();
+		let config: Result<ExecConfig>;
+		loop {
+			let target_path = curr_dir.join(exec_file);
+			let target_ref = target_path.as_path();
+			if target_ref.exists() {
+				config = ExecConfig::from_file(target_ref)
+					.context("Failed to load the exec config. Please make sure the exec config file is in the provided location and is formatted correctly.");
+				break;
+			}
+			if let Some(parent) = curr_dir.parent() {
+				curr_dir = parent.to_path_buf();
+			} else  {
+				log::info!("Using a default Exec Config");
+				config = ExecConfig::default();
+				break;
+			}
+		}
+
+		config
+	}.unwrap();
+
+	let plugin_data = exec_config.plugin_data;
+
 	let plugin_executor = PluginExecutor::new(
-		/* max_spawn_attempts */ 3,
-		/* max_conn_attempts */ 5,
+		/* max_spawn_attempts */ plugin_data.max_spawn.attempts,
+		/* max_conn_attempts */ plugin_data.max_conn.attempts,
 		/* port_range */ 40000..u16::MAX,
-		/* backoff_interval_micros */ 100000,
-		/* jitter_percent */ 10,
+		/* backoff_interval_micros */ plugin_data.backoff.micros,
+		/* jitter_percent */ plugin_data.jitter.percent,
+		/*grpc_buffer*/ plugin_data.grpc_buffer.size,
```

Comment:
Since all fields of `PluginExecutor` now come from the exec config file, I would also add a `ExecFile::get_plugin_executor() -> PluginExecutor ` function instead of creating it like this.